pkg/asset/installconfig/clusterid_test.go

```diff
@@ -0,0 +1,33 @@
+package installconfig
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_generateHumanID(t *testing.T) {
+	tests := []struct {
+		input string
+
+		expLen int
+	}{{
+		input:  "qwertyuiop",
+		expLen: 10 + randomLen + 1,
+	}, {
+		input:  "qwertyuiopasdfghjklzxcvbnm",
+		expLen: maxBaseLen + randomLen + 1,
+	}, {
+		input:  "qwe.rty.@iop!",
+		expLen: 13 + randomLen + 1,
+	}}
+	for _, test := range tests {
+		t.Run("", func(t *testing.T) {
+			got := generateHumanID(test.input)
+			t.Log("HumanID", got)
+			assert.Equal(t, test.expLen, len(got))
+			assert.Equal(t, regexp.MustCompile("[^A-Za-z0-9-]").ReplaceAllString(got, "-"), got)
```

Comment:
I would rather see the expected replacements written out explicitly in the test cases. This just ends up running the same code that is used in `generateHumanID`. If there is a bug in the replacement, this won't capture the bug.

Reply:
I tried to make it more meaningful with https://github.com/openshift/installer/pull/1280/commits/9f5c0c00348230fdf601bb466b2793638e852a8b -> https://github.com/openshift/installer/pull/1280/commits/4650c151c932b38341c39cdf71658eed1e695679 [compare](https://github.com/openshift/installer/compare/79a821c2088b46314d44913e7f51a8fbfe58e99f..653a9b2e8e9080b6729fae26eb2bb45dc7e232ce)